src/components/flat-table/flat-table-row-header/flat-table-row-header.component.tsx

```diff
@@ -71,8 +71,17 @@ export const FlatTableRowHeader = ({
     onKeyDown,
     isFirstCell,
     isExpandableCell,
+    highlighted,
+    selected,
+    setSelectedId,
   } = useCalculateStickyCells(internalId.current);
 
+  useEffect(() => {
+    if (isExpandableCell && (highlighted || selected)) {
+      setSelectedId?.(internalId.current);
```

Comment:
**comment**: could be a similar thing going wrong here to what I've called out in the `flat-table-cell` component. Maybe.